pkg/controller/pkg/revision/reconciler.go

```diff
@@ -121,52 +149,30 @@ func WithParserBackend(p parser.Backend) ReconcilerOption {
 }
 
 // WithLinter specifies how the Reconciler should parse a package.
-func WithLinter(l Linter) ReconcilerOption {
+func WithLinter(l parser.Linter) ReconcilerOption {
 	return func(r *Reconciler) {
 		r.linter = l
 	}
 }
 
-// BuildMetaScheme builds the default scheme used for identifying metadata in a
-// Crossplane package.
-func BuildMetaScheme() (*runtime.Scheme, error) {
-	metaScheme := runtime.NewScheme()
-	if err := pkgmeta.SchemeBuilder.AddToScheme(metaScheme); err != nil {
-		return nil, err
-	}
-	return metaScheme, nil
-}
-
-// BuildObjectScheme builds the default scheme used for identifying objects in a
-// Crossplane package.
-func BuildObjectScheme() (*runtime.Scheme, error) {
-	objScheme := runtime.NewScheme()
-	if err := apiextensionsv1alpha1.AddToScheme(objScheme); err != nil {
-		return nil, err
-	}
-	if err := v1beta1.AddToScheme(objScheme); err != nil {
-		return nil, err
-	}
-	return objScheme, nil
-}
-
 // Reconciler reconciles packages.
 type Reconciler struct {
-	client       client.Client
-	hook         Hooks
-	objects      Establisher
-	podLogClient kubernetes.Interface
-	parser       parser.Parser
-	linter       Linter
-	backend      parser.Backend
-	log          logging.Logger
-	record       event.Recorder
+	client   client.Client
+	cache    xpkg.Cache
+	revision resource.Finalizer
+	hook     Hooks
+	objects  Establisher
+	parser   parser.Parser
+	linter   parser.Linter
+	backend  parser.Backend
```

Comment:
I wonder if `parser` is the right name for the crossplane-runtime package. It's not immediately obvious what it's a parser for (i.e. packages), and `parser.Parser` stutters. `xpkg` would make sense if it didn't conflict with the local package of that name. Perhaps `package.Parser`, `package.Linter`, and `package.Backend`?

Reply:
Yeah I struggled with this a bit. Similarly if it was named `package` we would have `package.Package`. I think `parser` actually makes sense for `Parser` (despite stuttering) and `Backend` (i.e. it the backend for a parser). `Linter` is linting the output of a parser, which would be a package so that is kind of strange. I guess we could name it `pkg`, which would make the general crossplane-runtime library `pkg` and the crossplane specific additions `xpkg`... If we make this change I will likely do so in follow-up PR